MinitexPDFProtocols/MinitexPDFProtocols.swift

```diff
@@ -9,67 +9,92 @@
 import Foundation
 import UIKit
 
-// the items that are marked required are needed to
-// recreate an annotation in PSPDFKit later
-@objc public class MinitexPDFAnnotation: NSObject, Codable {
-  // required
-  public let pageIndex: UInt
-  // required
-  public let type: String
-  // required
-  public let bbox: String
-  // required
-  public let rects: [String]
 
-  // optional
-  public let color: String?
-  // optional
-  public let opacity: Float?
-  // optional
-  // we store this in case the field parsing fails, then we have the original JSON
-  public let JSONData: Data?
+/// A generic annotation in a PDF document.
+@objc public protocol MinitexPDFAnnotation: class {
 
-  // optional
-  // this value cannot be set in PSPDFKit
-  //var v: Int?
+  /// An annotation must belong to one page of the PDF document.
+  var page: MinitexPDFPage { get }
 
-  public init(pageIndex: UInt, type: String, bbox: String, rects: [String],
-              color: String?, opacity: Float?, JSONData: Data?) {
-    self.pageIndex = pageIndex
-    self.type = type
-    self.bbox = bbox
-    self.rects = rects
-    self.color = color
-    self.opacity = opacity
-    self.JSONData = JSONData
-  }
+  /// Complex annotations may be fully described by something like JSON from a particular
+  /// renderer. This allows easy access by the host to a version it can store or sync.
+  var serializedRepresentation: Data { get }
 }
 
-
+/// The View Controller for the PDF document.
 @objc public protocol MinitexPDFViewController: class {
-  // we have to pass in a dictionary because @objc protocol function
```

Comment:
Addressing just the comment here: There must have been an unrelated build error (it can and does happen often), since there is not an issue typically to do this, (and this init has been updated to do just that).